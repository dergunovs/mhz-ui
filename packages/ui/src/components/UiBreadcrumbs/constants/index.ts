export const LINKS = [
  { url: '/', title: 'Main' },
  { url: '/category', title: 'Categories' },
  { url: '/cpu', title: 'CPU' },
];

export const DEFAULT_COLOR = 'default';
export const DEFAULT_ARIA_LABEL = 'Breadcrumb';
