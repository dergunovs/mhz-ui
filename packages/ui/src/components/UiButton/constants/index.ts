export const DEFAULT_LAYOUT = 'primary';
export const DEFAULT_TYPE = 'button';
export const DEFAULT_ICON = undefined;

export const DEFAULT_SLOT = 'Button label';
export const DEFAULT_ARIA_LABEL = 'Button';
