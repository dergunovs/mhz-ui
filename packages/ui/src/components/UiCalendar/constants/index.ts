import { ICalendarEvent } from '../interface';

export const EVENTS: ICalendarEvent<object>[] = [
  {
    id: 'id of event',
    start: new Date(),
    end: new Date(),
    title: '+',
    content: [{ id: 1, text: 'Text' }],
    color: 'linear-gradient(135deg, rgb(168, 60, 255) 36%, rgb(235, 0, 123) 69%, rgb(251, 185, 0) 100%)',
  },
  {
    id: 'id of event',
    start: new Date(),
    end: new Date(),
    title: '+',
    content: [{ id: 2, text: 'Text 2' }],
    color: 'linear-gradient(135deg, rgb(168, 60, 255) 41%, rgb(235, 0, 123) 71%, rgb(0, 134, 255) 100%)',
  },
];
