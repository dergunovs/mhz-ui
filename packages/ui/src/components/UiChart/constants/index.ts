export const LABELS = [
  '07.07 - 13.07',
  '14.07 - 20.07',
  '21.07 - 27.07',
  '28.07 - 03.08',
  '04.08 - 10.08',
  '11.08 - 17.08',
  '18.08 - 24.08',
];

export const DATASETS = [
  {
    data: [28, 16, 12, 4, 22, 16, 14],
    label: 'Грудь',
    borderColor: '#00CED1',
    backgroundColor: '#00CED1',
  },
  {
    data: [4, 6, 0, 2, 2, 0, 6],
    label: 'Ноги',
    borderColor: '#000080',
    backgroundColor: '#000080',
  },
  {
    data: [22, 18, 8, 10, 25, 18, 16],
    label: 'Плечи',
    borderColor: '#C41E3A',
    backgroundColor: '#C41E3A',
  },
  {
    data: [0, 6, 0, 6, 5, 4, 2],
    label: 'Пресс',
    borderColor: '#32CD32',
    backgroundColor: '#32CD32',
  },
  {
    data: [32, 24, 12, 10, 33, 24, 20],
    label: 'Руки',
    borderColor: '#DA70D6',
    backgroundColor: '#DA70D6',
  },
  {
    data: [16, 18, 8, 12, 26, 22, 20],
    label: 'Спина',
    borderColor: '#FF8C00',
    backgroundColor: '#FF8C00',
  },
];

export const TYPE = 'Line';
export const TITLE = 'Chart title';
