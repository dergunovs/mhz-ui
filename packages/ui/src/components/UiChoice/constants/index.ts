export const options = [1, 2, 3, 4, 5];
export const modelValue = options[2];
export const title = 'Заголовок';
export const isTall = true;
export const isInput = true;
