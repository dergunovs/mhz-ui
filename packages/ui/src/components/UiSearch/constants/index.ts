export const DEBOUNCE_TIME = 300;

export const MODEL_VALUE = '';
export const LANG = 'ru';
export const IS_SUCCESS = true;

export const RESULTS = {
  products: [
    { _id: '1', title: 'AMD Ryzen 7700X' },
    { _id: '2', title: 'AMD Ryzen 7900X' },
  ],
  manufacturers: [{ _id: '1', title: 'AMD' }],
};

export const SEARCH_SCHEME = [
  { type: 'products', labels: ['title'], url: '/products' },
  { type: 'manufacturers', labels: ['title'], url: '/manufacturers' },
];
