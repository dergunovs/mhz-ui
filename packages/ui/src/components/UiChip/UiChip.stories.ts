import { Meta, StoryObj } from '@storybook/vue3';
import { html } from 'mhz-helpers';

import { DEFAULT_SLOT } from './constants';

import { UiChip } from '@/components';

const meta: Meta<typeof UiChip> = {
  component: UiChip,
  args: {},
  argTypes: {
    type: {
      options: ['default', 'success', 'error'],
    },
  },
};

export default meta;

export const Primary: StoryObj<typeof UiChip> = {
  render: (args) => ({
    components: { UiChip },
    setup: () => ({ args, DEFAULT_SLOT }),

    template: html` <div style="display:flex"><UiChip v-bind="args">{{DEFAULT_SLOT}}</UiChip></div>`,
  }),
};
