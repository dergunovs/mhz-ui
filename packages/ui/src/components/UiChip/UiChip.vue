<template>
  <div :class="$style.chip" :data-type="props.type" data-test="ui-chip">
    <slot></slot>

    <button v-if="props.isEdit" @click="emit('edit')" type="button" :class="$style.edit" data-test="ui-chip-edit">
      <IconEdit width="18" height="18" />
    </button>

    <UiClose v-if="props.isDelete" @click="emit('delete')" isSmall isDelete data-test="ui-chip-delete" />
  </div>
</template>

<script setup lang="ts">
import UiClose from '../UiClose/UiClose.vue';

import IconEdit from './icons/edit.svg';

import { DEFAULT_TYPE } from './constants';

interface IProps {
  type?: 'default' | 'success' | 'error';
  isEdit?: boolean;
  isDelete?: boolean;
}

interface IEmit {
  edit: [];
  delete: [];
}

const props = withDefaults(defineProps<IProps>(), {
  type: DEFAULT_TYPE,
});

const emit = defineEmits<IEmit>();
</script>

<style module lang="scss">
.chip {
  display: flex;
  flex-wrap: wrap;
  gap: 4px;
  align-items: center;
  height: 30px;
  padding: 4px 8px 3px;
  font-size: 0.875rem;
  color: var(--color-black);
  background-color: var(--color-gray-light);
  border-bottom: 1px solid var(--color-gray-dark);
  border-radius: 8px;

  &[data-type='success'] {
    color: var(--color-success-dark);
  }

  &[data-type='error'] {
    color: var(--color-error-dark);
  }
}

.edit {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 22px;
  height: 22px;
  padding: 0;
  color: var(--color-black);
  cursor: pointer;
  -webkit-user-select: none;
  user-select: none;
  background: none;
  border: none;
  -webkit-tap-highlight-color: transparent;
  -webkit-touch-callout: none;
}

:global(.dark) {
  .chip {
    background-color: var(--color-gray-dark);
    border-bottom: 1px solid var(--color-gray-dark-extra);
  }
}
</style>
