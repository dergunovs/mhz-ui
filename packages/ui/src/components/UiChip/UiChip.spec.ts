import { describe, it, expect, beforeEach, afterEach } from 'vitest';
import { VueWrapper, enableAutoUnmount } from '@vue/test-utils';
import { dataTest } from 'mhz-helpers';

import UiChip from './UiChip.vue';

import { DEFAULT_TYPE, DEFAULT_SLOT } from './constants';

import { wrapperFactory } from '@/test';

const chip = dataTest('ui-chip');
const chipEdit = dataTest('ui-chip-edit');
const chipDelete = dataTest('ui-chip-delete');

let wrapper: VueWrapper<InstanceType<typeof UiChip>>;

beforeEach(() => {
  wrapper = wrapperFactory(UiChip, { type: DEFAULT_TYPE }, { default: DEFAULT_SLOT });
});

enableAutoUnmount(afterEach);

describe('UiChip', async () => {
  it('exists', async () => {
    expect(wrapper.findComponent(UiChip)).toBeTruthy();
  });

  it('matches snapshot', async () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('shows text by slot', async () => {
    expect(wrapper.find(chip).text()).toBe(DEFAULT_SLOT);
  });

  it('sets type by props', async () => {
    expect(wrapper.find(chip).attributes('data-type')).toBe(DEFAULT_TYPE);
  });

  it('shows edit button and emits event by click', async () => {
    expect(wrapper.find(chipEdit).exists()).toBe(false);

    await wrapper.setProps({ isEdit: true });

    expect(wrapper.find(chipEdit).exists()).toBe(true);

    await wrapper.find(chipEdit).trigger('click');

    expect(wrapper.emitted('edit')).toHaveLength(1);
    expect(wrapper.emitted('edit')?.[0]).toEqual([]);
  });

  it('shows delete button and emits event by click', async () => {
    expect(wrapper.find(chipDelete).exists()).toBe(false);

    await wrapper.setProps({ isDelete: true });

    expect(wrapper.find(chipDelete).exists()).toBe(true);

    await wrapper.find(chipDelete).trigger('click');

    expect(wrapper.emitted('delete')).toHaveLength(1);
    expect(wrapper.emitted('delete')?.[0]).toEqual([]);
  });
});
