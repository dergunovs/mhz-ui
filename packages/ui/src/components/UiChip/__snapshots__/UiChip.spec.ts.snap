// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UiChip > matches snapshot 1`] = `"<div class="_chip_7be3f3" data-type="default" data-test="ui-chip">Chip text</div>"`;
